lookup_plugins/git_tag_prefix.py

```diff
@@ -0,0 +1,73 @@
+# python 3 headers, required if submitting to Ansible
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+lookup: git_tag_prefix
+author: Johnathan Kupferer <jkupfere@redhat.com>
+version_added: "2.9"
+short_description: Resolve git tag prefix to find latest version.
+description:
+- This lookup checks a git repository and returns a list of tags in the repo matching a prefix.
+options:
+  _terms:
+    description: git tag prefix to match
+    required: True
+notes:
+- Only git repos of the form https://github.com/ORG/REPO are supported at this time.
+"""
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible.plugins.lookup import LookupBase
+from ansible.utils.display import Display
+
+display = Display()
+
+from packaging import version
+import re
+import requests
+
+class LookupModule(LookupBase):
+
+    def run(self, terms, git_repo=None, variables=None, **kwargs):
+
+        # lookups in general are expected to both take a list as input and output a list
+        # this is done so they work with the looping construct 'with_'.
+        ret = []
+        for term in terms:
+            display.debug("File lookup term: %s" % term)
+
+            if not git_repo:
+                raise AnsibleError('git_repo must be passed to git_tag_prefix lookup')
+
+            match = re.match(r'https://([^/]+)/(.*)', git_repo)
+            if not match:
+                raise AnsibleError('sorry, cannot handle git repo url format {}'.format(git_repo))
+
+            git_host = match[1]
+            git_path = match[2]
+
+            if git_host == 'github.com':
```

Comment:
Why would we want to support only github ?

I think using the github API is the wrong way to approach this.

Can't we use standard Git to fetch the possible refs ?

Reply:
Git fetch is a much more expensive, more complex, and slower operation. I think the costs of implementing this with fetch outweigh the benefits, particularly because there are no practical benefits for us? We could later add logic to fallback to using a git fetch when repository type is unknown?